src/java/com/unifina/signalpath/ModuleWithUI.java

```diff
@@ -204,7 +205,7 @@ public void initialize() {
 
 			// User must have write permission to related Canvas in order to write to the UI channel
 			PermissionService permissionService = Holders.getApplicationContext().getBean(PermissionService.class);
-			if (!permissionService.canWrite(user, stream.getUiChannelCanvas())) {
+			if (!permissionService.check(user, stream.getUiChannelCanvas(), Permission.Operation.CANVAS_EDIT)) {
```

Comment:
Now that `permissionService.check` encapsulates the logic for the synthetic UI channel permissions, it **should** be possible to check the "right thing" here:

```suggestion
			if (!permissionService.check(user, stream, Permission.Operation.STREAM_PUBLISH)) {
```